app/views/regapps/_form.html.erb

```diff
@@ -1,6 +1,6 @@
 <div class="js-regapps">
-<% if current_user %>
-  <%= form_for [@user, Regapp.new], remote: true do |f| %>
+<% if user %>
```

Comment:
You shouldn't have to check if the user exists here, it's your obligation to set up a user local for this partial whenever you include it in a page.